webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -16,30 +16,63 @@ import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
 import { defaultExperimentColumns } from './glide-table/columns';
 import { Error, Loading, NoExperiments, NoMatches } from './glide-table/exceptions';
 import GlideTable, { SCROLL_SET_COUNT_NEEDED } from './glide-table/GlideTable';
+import { isValidSort, Sort, ValidSort } from './glide-table/MultiSortMenu';
 import TableActionBar from './glide-table/TableActionBar';
 import { useGlasbey } from './useGlasbey';
 
 interface Props {
   project: Project;
 }
 
+const makeSortString = (sorts: ValidSort[]): string =>
+  sorts.map((s) => `${s.column}=${s.direction}`).join(',');
+
 export const PAGE_SIZE = 100;
 const F_ExperimentList: React.FC<Props> = ({ project }) => {
   const [searchParams, setSearchParams] = useSearchParams();
 
-  const [page, setPage] = useState(
+  const [page, setPage] = useState(() =>
     isFinite(Number(searchParams.get('page'))) ? Number(searchParams.get('page')) : 0,
   );
-  const [experiments, setExperiments] = useState<Loadable<ExperimentItem>[]>(
+  const [sorts, setSorts] = useState<Sort[]>(() => {
+    const sortString = searchParams.get('sort') || '';
+    if (!sortString) {
+      return [{ column: undefined, direction: undefined }];
+    }
+    const components = sortString.split(',');
+    return components.map((c) => {
+      const [column, direction] = c.split('=', 2);
+      return {
+        column,
+        direction: direction === 'asc' || direction === 'desc' ? direction : undefined,
+      };
+    });
+  });
+  const [sortString, setSortString] = useState<string>('');
+  const [experiments, setExperiments] = useState<Loadable<ExperimentItem>[]>(() =>
     Array(page * PAGE_SIZE).fill(NotLoaded),
   );
   const [total, setTotal] = useState<Loadable<number>>(NotLoaded);
 
   useEffect(() => {
-    setSearchParams({ page: String(page) });
+    setSearchParams((params) => {
+      if (page) {
+        params.set('page', page.toString());
+      } else {
+        params.delete('page');
+      }
+      const sortString = makeSortString(sorts.filter(isValidSort));
```

Comment:
question: why is `sortString` being defined here instead of coming from the useState you set up above?

Reply:
sorry -- that's a holdover from when sortString was calculated on the fly. removing it now.